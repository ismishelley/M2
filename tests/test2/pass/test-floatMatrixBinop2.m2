int main() {
  float a;
  matrix float [2][3] M;
  a = 9.0;
  M = [[0.0,0.0,0.0];
      [0.0,0.0,0.0]];

  printStr("Matrix M is");
  myprint(M);
  printStr("Float a is");
  printFloat(a);
  printStr("");

  printStr("a * M is");
  myprint(a * M);    


  return 0;
}

void myprint(matrix float [2][3] M){
  int i;
  int j;
  for(i = 0; i < M:rows; ++i){
    for(j = 0; j < M:cols; ++j){
      printFloat(M[i][j]);
    }
    printStr("");
  }
}

