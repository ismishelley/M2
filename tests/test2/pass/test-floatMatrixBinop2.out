Matrix M is
0.000000	0.000000	0.000000	
0.000000	0.000000	0.000000	
Float a is
9.000000	
a * M is
0.000000	0.000000	0.000000	
0.000000	0.000000	0.000000	